File: python_study/220710_01_experiment_pass_by_reference/animal.py
class Animal:
    def __init__(self, name: str, size: int) -> None:
        self.name = name 
        self.size = size
    
    def handle_from_inside(self) -> None:
        print(f"Calling from inside handle_from_inside: \t{hex(id(self))}")

File: java_study/udemy-tim-buchalka/220710_02_experiment_passed_by_reference/myproject/Animal.java
package myproject;

public class Animal {

    private String name;
    private int size;
    
    public Animal(String name, int size) {
        this.name = name;
        this.size = size;
    }

    public String getName() {
        return this.name;
    }

    public int getSize() {
        return this.size;
    }

    public void handleFromInside(){
        System.out.println("Calling from inside handleFromInside(): \t" + this);
    }

}
